test/gtest/CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+include(googletest) #include googletest.cmake
+enable_testing()
+include(GoogleTest)
```

Comment:
We could move `include(GoogleTest)` inside cmake/googletest.cmake?

Reply:
include(googletest) and include(GoogleTest) do look a bit of confusing. Moving to googletest.cmake indeed makes the code cleaner IMO. 

One thing is the following line in this CmakeLists.txt file is dependent on include(GoogleTest). Not sure if moving include(GoogleTest) to a different file makes dependency/future reading complicated. 

ln16:  gtest_discover_tests(test_${TEST_NAME})

@JehandadKhan could you also provide insights? Thanks

Reply:
I think I agree with Bibek, we can make it cleaner by moving the additional lines